src/Kvasir/Reconstitution/Lookups.cs

```diff
@@ -27,22 +27,25 @@ internal static partial class Lookup {
         ///   The <see cref="DataExtractionPlan.ExpectedSource"/> of <paramref name="extractor"/> is the dynamic type
         ///   of each item in <paramref name="entities"/>, or is a base class or interface thereof.
         /// </pre>
-        public static object ByKey(DBData key, IEnumerable<object> entities, DataExtractionPlan extractor) {
+        public static T ByKey<T>(DBData key, IEnumerable<T> entities, DataExtractionPlan extractor) {
```

Comment:
I need to think about this a second - but I'm pretty sure this isn't going to work. The context in which we're going to be calling this is going to be fully based on reflection, which means we won't have the type available at compile time. I think we'll want to keep this as `object`-based and then use `extractor.ExpectedSource` as our type indicator.

(The tests right now are passing because they _do_ know the type of the objects at compile time. In practice, we're going to have something like a map of `Type` instances to lists of instances of that type, and we'll do a map lookup to get the collection of possible resolution targets.)

Reply:
Fixed!